### PR TITLE
hhob-lcjira: Open-source proposal, May 8

### PR DESCRIPTION
**Pull request by luisjira**
# Assignment Proposal

## Title
Implementing a CI/CD Pipeline for PGPro

## Names and KTH ID

  - Harald Hobbelhagen (hhob@kth.se)
  - Luis Jira (lcjira@kth.se)

## Deadline
- Task 3

## Category
- Open source

## Description
[PGPro][1] is the only open-source app for PGP encryption/decryption on iOS.

With this contribution we plan to implement a comprehensive CI/CD pipeline for the project.
The pipeline will include:
- Lint Checks
- Unit Tests
- UI Tests
- Deployment to the Apple AppStore
- Screenshot generation

**Relevance**
We will implement automated testing in combination with continuous deployment, which are central topics of this course in weeks 2 and 3.

[1] https://github.com/lucanaef/PGPro

### CODE REVIEW

**Comment by Deee92** [comment]
Hi @luisjira, thanks for the proposal!
PGPro is not strictly a DevOps-related project. The task you propose is very DevOps, though.
Also, PGPro does not seem to have been very active recently. Can you open an issue in the repo proposing that you will work on implementing a CI/CD pipeline for them?

**Comment by Deee92** [comment]
@luisjira 🔔

**Comment by haraldhob** [comment]
@Deee92 I opened now an issue.

https://github.com/lucanaef/PGPro/issues/113

**Comment by haraldhob** [comment]
@Deee92 we got a positive reply

**Comment by Deee92** [comment]
Hi @luisjira, that's great! Good luck. :)

File: contributions/opensource/hhob-lcjira/README.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+## Title
+Implementing a CI/CD Pipeline for PGPro
+
+## Names and KTH ID
+
+  - Harald Hobbelhagen (hhob@kth.se)
+  - Luis Jira (lcjira@kth.se)
+
+## Deadline
+- Deadline 3
```
Comment (Deee92) [inline contributions/opensource/hhob-lcjira/README.md]: @luisjira, please change line 12 to `- Task 3`

**Comment by Deee92** [comment]
Hi @luisjira, please fetch upstream, so that this PR passes checks and we can merge it.

**Pull request merged**